src/devices/MAX31856/MAX31856.cs

```diff
@@ -0,0 +1,201 @@
+    /// Add documentation here
+    /// </summary>
+    public class MAX31856 : IDisposable
+    {
+        private readonly byte _thermocoupleType;
+        private SpiDevice _spiDevice;
+
+        #region SpiSettings
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int SpiClockFrequency = 5_000_000;
+
+        /// <summary>
+        /// Spi Clock Frequency from the Technical Manual of the device
+        /// </summary>
+        public const int DataFlow = 0;
+
+        /// <summary>
+        /// MAX31856 SPI Mode
+        /// </summary>
+        public const SpiMode SpiMode = System.Device.Spi.SpiMode.Mode1;
+
+        #endregion
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature GetTemperature() => Temperature.FromDegreesCelsius(ReadTemperature());
+
+        /// <summary>
+        ///  Reads the temperature from the sensor
+        /// </summary>
+        /// <returns>
+        ///  Temperature in degrees celsius
+        /// </returns>
+        public Temperature GetCJTemperature() => Temperature.FromDegreesCelsius(ReadCJTemperature());
+
+        /// <summary>
+        /// Creates a new instance of the max31856.
+        /// </summary>
+        /// <param name="spiDevice">The communications channel to a device on a SPI bus</param>
+        /// <param name="thermocoupleType">Thermocouple Type T,K,R</param>
+        public MAX31856(SpiDevice spiDevice, ThermocoupleType thermocoupleType)
+        {
+            _spiDevice = spiDevice ?? throw new ArgumentNullException(nameof(spiDevice));
+            _thermocoupleType = (byte)thermocoupleType;
+            Initialize();
+        }
+
+        #region private and internal
+
+        /// <summary>
+        /// Standard initialization routine.
+        /// </summary>
+        /// /// <remarks>
+        /// You can add new write lines if you want to alter the settings of the device. Settings can be found in the Technical Manual
+        /// </remarks>
+        private void Initialize()
+        {
+            Span<byte> configurationSetting0 = stackalloc byte[]
+            {
+                (byte)Register.WRITE_CR0,
+                0x90
+            };
+            Span<byte> configurationSetting1 = stackalloc byte[]
+            {
+                (byte)Register.WRITE_CR1,
+                _thermocoupleType
+            };
+
+            Write(configurationSetting0);
+            Write(configurationSetting1);
+        }
+
+        /// <summary>
+        /// Command to Get Temperature from the Device
+        /// </summary>
+        /// <remarks>
+        /// Initializes the device and then reads the data next it converts the bytes to thermocouple temperature  and throws an exception if there is an error
+        /// </remarks>
```

Comment:
Bonus: You can use the '/// <exception'  to describe the exception

Reply:
I added this but am not sure if I am implementing it correctly.